processor/otel/consumer.go

```diff
@@ -428,12 +443,50 @@ func parseSpan(span *tracepb.Span, event *model.Span) {
 		}
 	}
 
-	if destination != (model.Destination{}) {
-		event.Destination = &destination
+	if http.URL != nil && (destination.Address == nil || destination.Port == nil) {
+		if fullURL, err := url.Parse(*http.URL); err == nil {
+			url := url.URL{Scheme: fullURL.Scheme, Host: fullURL.Host}
+			hostname := truncate(url.Hostname())
+			var port int
+			portString := url.Port()
+			if portString != "" {
+				port, _ = strconv.Atoi(portString)
+			} else {
+				port = schemeDefaultPort(url.Scheme)
+			}
+
+			// Set destination.{address,port} from the HTTP URL,
+			// if no peer.* attributes were specified.
+			if destination.Address == nil {
+				destination.Address = &hostname
+			}
+			if destination.Port == nil && port > 0 {
+				destination.Port = &port
+			}
```

Comment:
This might lead to `destination.Address` being set by `peer.*`  and `destination.Port` being derived from `http.URL` and vice versa - not sure if this is a real world concern or not, but might be confusing.

Reply:
This is mostly to cater for taking the scheme default port. I suppose we could always set both `destination.address` and `destination.port` from `http.url` if provided. Do you think that would be less confusing?

Reply:
As that would be in line with what we do in our own agents, I'll go ahead and make that change.